build/Common.props

```diff
@@ -10,6 +10,11 @@
     <ImplicitUsings>enable</ImplicitUsings>
     <!--temporarily disable. See 3958-->
     <!--<AnalysisLevel>latest-All</AnalysisLevel>-->
+    <ExposeExperimentalFeatures Condition="'$(ExposeExperimentalFeatures)' == ''">true</ExposeExperimentalFeatures>
```

Comment:
Rename this property to something different than the build property?

Reply:
Not sure what you mean, this _is_ the build property 😄 Putting `Condition="'$(ExposeExperimentalFeatures)' == ''"` here just allows the build to override it.

Reply:
Could we get rid of this by updating the below introduced `PropertyGroup` to this?

```xml
<PropertyGroup Condition= "'$(ExposeExperimentalFeatures)' == ''" OR "'$(ExposeExperimentalFeatures)' == 'true'">
    <DefineConstants>$(DefineConstants);EXPOSE_EXPERIMENTAL_FEATURES</DefineConstants>
</PropertyGroup>
```

Reply:
We could yes but we lose then the ability to easily toggle it on/off when working in the IDE. Being able to just toggle a true/false IMO is nice so you can work through (potential) CI failures in one of the paths.

Reply:
> We could yes but we lose then the ability to easily toggle it on/off when working in the IDE.

How would this be any different from what the PR currently has? One the IDE, by default the experimental path would be selected as `ExposeExperimentalFeatures` would be unset. If one wants to turn it off they would have to specify the property in build options on the IDE.

Reply:
What I was thinking for IDE if you want to run the "stable" style build you just change this to `false` (locally): https://github.com/open-telemetry/opentelemetry-dotnet/blob/4654de745489ae08ff41ad2675c55fac2dd32b8c/build/Common.props#L13

Reply:
Got it. Makes sense.